resolwe_bio/processes/clustering/hierarchical_clustering_etc.py

```diff
@@ -0,0 +1,316 @@
+    else:
+        excluded = sorted(outer.index.difference(inner.index))
+    return inner, excluded
+
+
+class ClusterTimeCourse(Process):
+    """Cluster gene expression time courses.
+
+    Hierarchical clustering of expression time courses.
+    """
+
+    slug = "clustering-hierarchical-etc"
+    name = "Hierarchical clustering of time courses"
+    process_type = "data:clustering:hierarchical:etc"
+    version = "1.0.0"
+    scheduling_class = SchedulingClass.INTERACTIVE
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {"docker": {"image": "resolwebio/common:1.6.0"}},
+        "resources": {"cores": 1, "memory": 8192},
+        "relations": [{"type": "series"}],
+    }
+    data_name = "Hierarchical clustering of time courses"
+
+    class Input:
+        """Input fields to process ClusterTimeCourse."""
+
+        expressions = ListField(
+            DataField("expression"),
+            relation_type="series",
+            label="Time series relation",
+            description="Select time course to which the expressions belong to.",
+        )
+        genes = ListField(
+            StringField(),
+            label="Gene subset",
+            required=False,
+            description="Select at least two genes or leave this field empty.",
+        )
+        gene_species = StringField(
+            label="Species",
+            description="Species to which the selected genes belong to. "
+            "This field is required if gene subset is set.",
+            required=False,
+            hidden="!genes",
+            allow_custom_choice=True,
+            choices=[
+                ("Dictyostelium discoideum", "Dictyostelium discoideum"),
+                ("Homo sapiens", "Homo sapiens"),
+                ("Macaca mulatta", "Macaca mulatta"),
+                ("Mus musculus", "Mus musculus"),
+                ("Rattus norvegicus", "Rattus norvegicus"),
+            ],
+        )
+        gene_source = StringField(
+            label="Gene ID database of selected genes",
+            description="This field is required if gene subset is set.",
+            required=False,
+            hidden="!genes",
+        )
+        distance = StringField(
+            label="Distance metric",
+            choices=[
+                ("spearman", "Spearman"),
+                ("pearson", "Pearson"),
+            ],
+            default="spearman",
+        )
+        linkage = StringField(
+            label="Linkage method",
+            choices=[
+                ("average", "average"),
+                ("single", "single"),
+                ("complete", "complete"),
+            ],
+            default="average",
+        )
+        ordering = BooleanField(
+            label="Use optimal ordering",
+            description="Results in a more intuitive tree structure, "
+            "but may slow down the clustering on large datasets",
+            default=False,
+        )
+
+    class Output:
```

Comment:
If we propagate `source`, `species`, `build` and `feature_type` from expression outputs, we can use this process as a selector of interesting genes, similar to what we do after [differential expression](https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/processes/differential_expression/deseq.py#L253) on volcano plot.

Reply:
Thanks will add this to the outputs.